ui: CSS modules for SqlBox components

koorosh opened the pull request:
Depends on #48155
Related to #47527

Refactor following components to use
CSS modules instead of global styles:
- SqlBox
- highlight

Note, `hljs-` classes are 3rd-party
styles which are overriden with custom
styles.

Release note: None

koorosh committed:
ui: CSS modules for Statements page filters

This change refactors following components to
use CSS modules instead of styles defined as
global:
- Dropdown
- Search
- PageConfig

Release note: None

koorosh committed:
ui: CSS modules for SummaryCard component

Release note: None

koorosh committed:
ui: CSS modules for Tooltip shared component

Release note: None

koorosh committed:
ui: CSS modules for Button component

Before, Button component had styles defined with
own namespacing and now styles are imported as
CSS module.

The main change is separation BackIcon component
which was coupled with Button and the way
custom styles were applied to display button as
a link without highlighting.

- BackIcon is refactored to it's own module with
styles
- custom styles are applied with new Button type
option - `unstyled-link`.

Release note: None

koorosh committed:
ui: CSS modules for Statement Details page

Following components are refactored to use CSS modules
 instead of global styles:
- Statement Details page
- Statements Diagnostics components
- Anchor
- Text
- Empty

Before, `Empty` component had styles which override
 Text, and Anchor components. Now, these styles are
 passed via `className` component props and custom
 styles defined by outer component.

Release note: None

koorosh committed:
ui: CSS modules for Statement Details page

Following components are refactored to use CSS modules
 instead of global styles:
- Statement Details page
- Statements Diagnostics components
- Anchor
- Text
- Empty

Before, `Empty` component had styles which override
 Text, and Anchor components. Now, these styles are
 passed via `className` component props and custom
 styles defined by outer component.

Release note: None

koorosh committed:
ui: CSS modules for SqlBox components

Refactor following components to use
CSS modules instead of global styles:
- SqlBox
- highlight

Note, `hljs-` classes are 3rd-party
styles which are overriden with custom
styles.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/48217)
<!-- Reviewable:end -->

koorosh committed:
ui: Storybook for Statements Details page

Statement Details pages has four tabs which aren't
controlled and it wasn't possible to open particular
tab and render it in storybook. It is also led to
usability issue where selected tab was not preserved
after page refresh.

With current commit:
- selected tab is tracked as search param in URL
- added storybook for Statement Details page for
every tab except Diagnostics tab.
Content of Diagnostics tab is independent connected
view and has to be defined in it's own storybook.

Release note (admin ui change): preserve selected tab in Statement
Details page after reload.

koorosh committed:
ui: Bind CSS class names with classnames/bind

Instead of accessing to CSS modules class names as fields,
`classnames/bind` function is used to bind class names to
hashed names in imported CSS module file.

Release note: None

koorosh commented:
bors r+
